### PR TITLE
Cbapi 4911 v7 unit tests

### PR DESCRIPTION
**Pull request by heiderje-vmware**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ x] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ x] Tests have been added that prove the fix is effective or that the feature works.
- [ x] New and existing tests pass locally with the changes.
- [ x] Code follows the style guidelines of this project (PEP8, clean code, linter).
- [x ] A self-review of the code has been done.

## What is the ticket or issue number?
[CPAPI_4911](https://jira.carbonblack.local/browse/CBAPI-4911)

## Does this introduce a breaking change?

- [ ] Yes
- [ x] No

### CODE REVIEW

**Review by avanbrunt-cb — COMMENTED** [review]
What's the testing plan for covering more alert types and the mappings? I see this maps more closely to what tests we had before.

Also what is the plan for fixing the broken tests as the pipeline is still failing

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py [hunk 1]
```diff
@@ -0,0 +1,294 @@
...
+                                    "process_effective_reputation": "TRUSTED_WHITE_LIST",
+                                    "process_reputation": "TRUSTED_WHITE_LIST",
+                                    "process_cmdline": "\"powershell.exe\" & {1..10 | % { \n try { [System.IO.File]::\
+                                    Copy(\\\"\"\\\\?\\GLOBALROOT\\Device\\HarddiskVolumeShadowCopy$_\\Windows\\System32\
+                                    \\config\\SAM\\\"\" , \\\"\"$env:TEMP\\SAMvss$_\\\"\", \\\"\"true\\\"\") } catch {}\
+                                    \n ls \\\"\"$env:TEMP\\SAMvss$_\\\"\" -ErrorAction Ignore\n}}",
+                                    "process_username": "KOGNOS-W19-CB-3\\Administrator",
+                                    "process_issuer": ["Microsoft Windows Production PCA 2011"],
+                                    "process_publisher": ["Microsoft Windows"],
+                                    "parent_guid": "7DESJ9GN-006a07ff-000009a0-00000000-1d9b8d5c2fcbfb7",
```
Comment (kebringer-cb) [inline src/tests/unit/fixtures/platform/mock_alerts_v7.py:227]: Still four instances of org_key "7DESJ9GN" to be obfuscated.

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py [hunk 2]
```diff
@@ -0,0 +1,294 @@
...
+                                    "parent_reputation": "TRUSTED_WHITE_LIST",
+                                    "parent_cmdline": "\"c:\\windows\\system32\\windowspowershell\\v1.0\\powershell.exe\
+                                    \" -c \"cd c:\\ ; echo MYPID=$PID; Get-Date ; Invoke-AtomicTest T1003.002-6 \"",
+                                    "parent_username": "KOGNOS-W19-CB-3\\Administrator", "childproc_guid": "",
+                                    "childproc_username": "",
+                                    "childproc_cmdline": "", "ml_classification_final_verdict": "NOT_ANOMALOUS",
+                                    "ml_classification_global_prevalence": "MEDIUM",
+                                    "ml_classification_org_prevalence": "LOW"}
+
+GET_ALERT_RESP_WITH_NOTES = {"org_key": "VZMTPVJZR2",
```
Comment (kebringer-cb) [inline src/tests/unit/fixtures/platform/mock_alerts_v7.py:242]: org_key to be obfuscated in two places

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py [hunk 3]
```diff
@@ -0,0 +1,294 @@
...
+                                          "status": "OPEN"},
+                             "determination": {"change_timestamp": "2023-07-18T00:02:17.465Z", "value": "NONE",
+                                               "changed_by_type": "null",
+                                               "changed_by": "null"}, "tags": "null", "alert_notes_present": "false",
+                             "threat_notes_present": "false",
+                             "is_updated": "false", "mdr_alert": "false", "mdr_alert_notes_present": "false",
+                             "mdr_threat_notes_present": "false", "netconn_remote_port": 443,
+                             "netconn_local_port": 56802,
+                             "netconn_protocol": "TCP", "netconn_remote_domain": "westus3.monitoring.azure.com",
+                             "netconn_remote_ip": "40.64.132.128", "netconn_local_ip": "10.224.0.74",
```
Comment (kebringer-cb) [inline src/tests/unit/fixtures/platform/mock_alerts_v7.py:270]: obfuscate IP addresses

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py [hunk 4]
```diff
@@ -0,0 +1,294 @@
...
+                             "netconn_remote_ipv4": "40.64.132.128",
+                             "netconn_local_ipv4": "10.224.0.74", "k8s_cluster": "akozin:akozin-azure-cni",
```
Comment (kebringer-cb) [inline src/tests/unit/fixtures/platform/mock_alerts_v7.py:272]: obfuscate k8s_cluster

File: src/tests/unit/platform/test_alertsv7_api.py [hunk 5]
```diff
@@ -0,0 +1,872 @@
...
+
+@pytest.fixture(scope="function")
+def cbcsdk_mock(monkeypatch, cb):
+    """Mocks CBC SDK for unit tests"""
+    return CBCSDKMock(monkeypatch, cb)
+
+
+# ==================================== UNIT TESTS BELOW ====================================
+
+def test_query_alert_with_all_bells_and_whistles(cbcsdk_mock):
```
[the remark this answers is not in the document]
Comment (kebringer-cb) [reply src/tests/unit/platform/test_alertsv7_api.py:60]: Moved the request for time fields to CBAPI-4979.
Moved the request for exclusions to CBAPI-4992

File: src/cbc_sdk/platform/legacy_alerts.py [hunk 6]
```diff
@@ -1,4 +1,18 @@
...
+# *******************************************************
+# *
+# * DISCLAIMER. THIS PROGRAM IS PROVIDED TO YOU "AS IS" WITHOUT
+# * WARRANTIES OR CONDITIONS OF ANY KIND, WHETHER ORAL OR WRITTEN,
+# * EXPRESS OR IMPLIED. THE AUTHOR SPECIFICALLY DISCLAIMS ANY IMPLIED
+# * WARRANTIES OR CONDITIONS OF MERCHANTABILITY, SATISFACTORY QUALITY,
+# * NON-INFRINGEMENT AND FITNESS FOR A PARTICULAR PURPOSE.
+
+"""Legacy setters for alert v6 api search criteria"""
+
```
[the remark this answers is not in the document]
Comment (kebringer-cb) [reply src/cbc_sdk/platform/legacy_alerts.py:15]: Do not need to action here.
Done in PR #399

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py (hunk 1 above)
Comment (kebringer-cb) [reply src/tests/unit/fixtures/platform/mock_alerts_v7.py]: thanks

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py (hunk 2 above)
Comment (kebringer-cb) [reply src/tests/unit/fixtures/platform/mock_alerts_v7.py]: thanks

File: src/tests/unit/fixtures/platform/mock_alerts_v7.py (hunk 3 above)
Comment (kebringer-cb) [reply src/tests/unit/fixtures/platform/mock_alerts_v7.py]: fixed.

File: src/cbc_sdk/platform/alerts.py [hunk 7]
```diff
@@ -153,6 +143,16 @@ def __init__(self, cb, model_unique_id, initial_data=None):
 
     class Note(PlatformModel):
         """Represents a note within an alert."""
+        REMAPPED_NOTES_V6_TO_V7 = {
```
[the remark this answers is not in the document]
Comment (kebringer-cb) [reply src/cbc_sdk/platform/alerts.py:145]: Move to another ticket to review use of enumerations for validation, and correct as needed - CBAPI-5022

**Review by kebringer-cb — APPROVED** [review]
some items identified in comments have follow on tickets created.

**Review by avanbrunt-cb — APPROVED** [review]
Let's get all the tests merged and reassess the current testing state

**Pull request merged**